tests/aws-cpp-sdk-s3-unit-tests/S3UnitTests.cpp

```diff
@@ -16,8 +17,29 @@ using namespace Aws::S3::Model;
 
 const char* ALLOCATION_TAG = "S3UnitTest";
 
+class S3UnitTest_S3EmbeddedErrorTestNonOKResponse_Test;
+
+class S3TestClient : public S3Client
```

Comment:
This is a quick way of adding a test wrapper on top of S3 client class to be able to make unit test related changes